### ISSUE TITLE
False Positive Vertical Parameter Alignment on Call when closure is defined on new line

### ISSUE THREAD

**Issue by LeoNavel** [realm/SwiftLint#5338]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Vertical Parameter Alignment on Call rule is false positive when closure is defined on new line. Following code (taken from the last example of the [Non Triggering Examples](https://realm.github.io/SwiftLint/vertical_parameter_alignment_on_call.html)) should not trigger violation but it does.

```swift
UIViewPropertyAnimator.runningPropertyAnimator(
    withDuration: 2.0,
    delay: 0.0,
    options: [.curveEaseIn]
) {
    // animations
} completion: { _ in
    // completion
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'main.swift' (1/1)
/Users/User/main.swift:5:3: warning: Vertical Parameter Alignment on Call Violation: Function parameters should be aligned vertically if they're in multiple lines in a method call (vertical_parameter_alignment_on_call)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version: 0.53.0
* Installation method used: Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - vertical_parameter_alignment_on_call
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  No.
* Which Xcode version are you using (check `xcodebuild -version`)?
  Xcode 15.0.1
  Build version 15A507
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

Content of _main.swift_
```swift
UIViewPropertyAnimator.runningPropertyAnimator(
    withDuration: 2.0,
    delay: 0.0,
    options: [.curveEaseIn]
) { // 
    // animations
} completion: { _ in
    // completion
}
```

```bash
$ swiftlint lint --path main.swift --no-cache --enable-all-rules
warning: The --path option is deprecated. Pass the path(s) to lint last to the swiftlint command.
Linting Swift files at paths main.swift
Linting 'main.swift' (1/1)
warning: The `anyobject_protocol` rule is now deprecated and will be completely removed in a future release.
warning: The `inert_defer` rule is now deprecated and will be completely removed in a future release.
warning: The `unused_capture_list` rule is now deprecated and will be completely removed in a future release.
/Users/User/main.swift:2:19: warning: No Magic Numbers Violation: Magic numbers should be replaced by named constants (no_magic_numbers)
/Users/User/main.swift:5:3: warning: Vertical Parameter Alignment on Call Violation: Function parameters should be aligned vertically if they're in multiple lines in a method call (vertical_parameter_alignment_on_call)
Done linting! Found 2 violations, 0 serious in 1 file.
```

**Comment by LeoNavel** [comment]
It seems that #5332 resolved this issue yesterday.
